main.tf

```
@@ -644,7 +644,7 @@ resource "aws_elastic_beanstalk_environment" "default" {
   setting {
     namespace = "aws:autoscaling:launchconfiguration"
     name      = "SecurityGroups"
-    value     = join(",", compact(sort(concat([module.aws_security_group.id], var.associated_security_group_ids))))
+    value     = join(",", compact(sort(concat([module.aws_security_group.id == null ? "" : module.aws_security_group.id], var.associated_security_group_ids))))
```

@samcrudge I had a look at this again and @goruha 's suggestion appears to be:

```suggestion
    value     = join(",", sort(compact(concat([module.aws_security_group.id], var.associated_security_group_ids))))
```

Did you try this one?

---

Sorry i thought that was what i had committed! sorry!

---

no worries! 
And so this branch is working with `create_security_group = false` ?

---

Just doing some more testing on other projects we have blocked by this.

---

I have my projects working perfect with the suggested changes, thanks @joe-niland it was very helpful!

---

I'm using this line in my pipeline until this change is merged 

```
- name: Adjust beanstalk module PROD
      working-directory: ./environments/prod/.terraform/modules/beanstalk_production_environment
      run: sed -i 's|compact(sort(concat|sort(compact(concat|g' main.tf
```

---

@samcrudge it's good to go from our side. Have you completed your testing?

---

Yes I deployed to live environment and was successful

---

I've tested across 2 projects successfully now. @joe-niland